### ISSUE TITLE
Question about "Inverse Model" in the paper.

### ISSUE THREAD

**Issue by leiloong** [TimDettmers/ConvE#35]
I have understood how to detect inverse relations by in the "Inverse Model" in the paper. 
   But I do not quite get the idea of "k matches" in testing,
   "At test time, we check if the test triple has inverse matches outside the test set: if k matches are found, we sample a permutation of the top k ranks for these matches; if no match is found, we select a random rank for the test triple." 
   Is this means, if we have (s, r_i, o), we will find all  (o, r_j, s) in training set (r_j \in R), and k is the number of  (o, r_j, s) ? 
   If yes, why should we do so? 
   It seems this test looks like "relation prediction" instead of "entity prediction".
   Maybe I do not really understand it.

**Comment by TimDettmers** [comment]
If you have inverse relationships generalization/specialization and at test time you have the test triple `(Felidae, generalization, cat)` then you might have matches:

- `(Cat, specialization, Felidae)`
- `(Lion, specialization, Felidae)`
- `(Tiger, specialization, Felidae)` 

What I do is that I select a random order for these matches which means the correct triple might be anywhere in the top 3. Does that make sense?

**Comment by leiloong** [comment]
Got it, thanks a lot. 
Another small question,  are the results of the "Inverse Model" "raw" or "filtered" (remove other positive cases in the k matches) ?

**Comment by leiloong** [comment]
Thanks again, checked the code of inverse_model.py and find it is a filtered result.
